sympy/matrices/expressions/matpow.py

```diff
@@ -56,14 +56,25 @@ def doit(self, **kwargs):
             args = self.args
         base = args[0]
         exp = args[1]
-        if isinstance(base, MatrixBase) and exp.is_number:
+        if exp.is_zero and base.is_square:
+            try:
```

Comment:
I'm not sure why you're using a try/catch statement to evaluate an object of instance ```MatrixBase``` to the zero power. It's already defined here: https://github.com/sympy/sympy/blob/19f589457dfc4822e375e8715377ad71b4d21f25/sympy/matrices/matrices.py#L521

But it looks like all tests pass and this addresses the issue.